daemon/oci_linux.go

```diff
@@ -775,7 +783,12 @@ func (daemon *Daemon) createSpec(c *container.Container) (retSpec *specs.Spec, e
 	if err := setResources(&s, c.HostConfig.Resources); err != nil {
 		return nil, fmt.Errorf("linux runtime spec resources: %v", err)
 	}
-	s.Linux.Sysctl = c.HostConfig.Sysctls
+	// Make sure we don't override sysctls that we injected above.
+	for k, v := range c.HostConfig.Sysctls {
+		if _, ok := s.Linux.Sysctl[k]; !ok {
```

Comment:
Hmm, so, if a user have set `kernel.domainname` in HostConfig.Sysctls it will stop working, right? It seems a backward-incompatible change. Also, since setting sysctls directly is lower level than parameters in Config, it probably should take precedence.

Reply:
> Hmm, so, if a user have set kernel.domainname in HostConfig.Sysctls it will stop working, right?

I couldn't find anywhere that actually set `HostConfig.Sysctls` (though since we expose it through the API maybe somebody has used it in the past). But you're right that the precedence is the wrong way around (I forgot that `HostConfig` was actually accessible through the API -- if it was internal-only this change wouldn't break because I don't think we've ever used `Sysctls` internally).